src/App.tsx

```diff
@@ -1,24 +1,78 @@
 import React from 'react';
+import { clearInterval } from 'timers';
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
-
-export default App;
+type Props = {};
+
+interface State {
+  date: Date;
+  timerId: ReturnType<typeof setInterval>;
+  isClockVisible: boolean;
+}
+
+export class App extends React.Component<Props, State> {
+  state = {
+    date: new Date(),
+    timerId: setInterval(() => {}, 0),
+    isClockVisible: true,
+  };
+
+  componentDidMount() {
+    this.state.timerId = setInterval(
+      () => this.tick(),
+      1000,
+    );
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.state.timerId);
+  }
```

Comment:
Create a Clock component updating the time every second.

Use class component
Start the timer only when the component is added to the page (componentDidMount)
Update the state every second using setInterval
Save timerId to stop it later